### PR TITLE
Lens 781

### PR DESCRIPTION
**Pull request by hdhayneCoveo**
mainly this PR is just moving classes to packages and some renaming , except one bug fix (i added a comment near it)

all comments are most welcomed , also if you think we shouldn't do that or we should do it in smaller chunks, let me know

### CODE REVIEW

File: src/main/java/com/coveo/source/SourceClient.java [hunk 1]
```diff
@@ -162,7 +176,7 @@ public HttpResponse<String> deleteDocument(String sourceId, String documentId, B
     }
 
     /**
-     * Manage batches of items in a push source. See [Manage Batches of Items in a Push Source](https://docs.coveo.com/en/90)
+     * Manage batches of items in a push source. See [Manage Batches of Items in a Push SourceClient](https://docs.coveo.com/en/90)
```
Comment (samisayegh) [inline src/main/java/com/coveo/source/SourceClient.java:179]: Were the renames of `Source` -> `SourceClient` in this file intended?

File: src/main/java/com/coveo/pushclient/ApiStatus.java [hunk 2]
```diff
@@ -1,9 +1,9 @@
-package com.coveo.pushapiclient;
+package com.coveo.pushclient;
 
 /**
- * Enum for possible PushAPI statuses. See [Updating the Status of a Push Source](https://docs.coveo.com/en/35).
+ * Enum for possible PushAPI statuses. See [Updating the Status of a Push SourceClient](https://docs.coveo.com/en/35).
  */
-public enum PushAPIStatus {
+public enum ApiStatus {
```
Comment (samisayegh) [inline src/main/java/com/coveo/pushclient/ApiStatus.java:6]: Could `SourceStatus` be a better name for this enum? Is it the status of the api? or the source?

**Review by louis-bompart — CHANGES_REQUESTED** [review]
- [ ] Split bug fix in another PR
- [ ] LENS-781 mentions a technical document that documents the decision, if `@coveo/dx` retains ownership of the SDK, I cannot approve new features on the SDK until a more long-term plan addressing its maintainability is put in motion.

File: src/main/java/com/coveo/pushclient/ApiStatus.java [hunk 3]
```diff
@@ -1,9 +1,9 @@
-package com.coveo.pushapiclient;
+package com.coveo.pushclient;
 
 /**
- * Enum for possible PushAPI statuses. See [Updating the Status of a Push Source](https://docs.coveo.com/en/35).
+ * Enum for possible PushAPI statuses. See [Updating the Status of a Push SourceClient](https://docs.coveo.com/en/35).
```
Comment (samisayegh) [inline src/main/java/com/coveo/pushclient/ApiStatus.java:4]: Not sure about the `Source` -> `SourceClient` rename here

File: README.md [hunk 4]
```diff
@@ -19,22 +19,22 @@ Using Maven:
 See more examples in the `./samples` folder.
 
 ```java
-import com.coveo.pushapiclient.DocumentBuilder;
-import com.coveo.pushapiclient.Source;
+import com.coveo.document.DocumentBuilder;
+import com.coveo.source.SourceClient;
```
Comment (samisayegh) [inline README.md:23]: If these paths are changing, does that mean we are introducing a breaking change for the users of the sdk?

File: README.md [hunk 5]
```diff
@@ -19,22 +19,22 @@ Using Maven:
...
 
 import java.io.IOException;
 import java.net.http.HttpResponse;
 
 public class PushOneDocument {
     public static void main(String[] args) {
-        Source source = new Source("my_api_key", "my_org_id");
+        SourceClient sourceClient = new SourceClient("my_api_key", "my_org_id");
```
Comment (samisayegh) [inline README.md:30]: Same question here, this feels like a breaking change? Is it intentional? If so, is it necessary?

File: samples/CreateSource.java [hunk 6]
```diff
@@ -1,22 +1,24 @@
...
 
 import java.io.IOException;
 import java.net.http.HttpResponse;
 
 public class CreateSource {
     public static void main(String[] args) {
         PlatformUrl platformUrl = new PlatformUrlBuilder()
-                .withRegion(Region.AU) // If your organization is located in a different region than Region.US
+                .withEnvironment(Environment.STAGING)
+                .withRegion(Region.US) // If your organization is located in a different region than Region.US
```
Comment (y-lakhdar) [inline samples/CreateSource.java:15]: Why changing to the default region. The goal of that code snippet is to show user how to switch to another region (other than the default US)

File: samples/CreateSource.java [hunk 7]
```diff
@@ -1,22 +1,24 @@
...
                 .build();
-        Source source = new Source("my_api_key", "my_org_id", platformUrl);
+        SourceClient sourceClient = new SourceClient("my_api_key", "my_org_id", platformUrl);
         try {
-            HttpResponse<String> response = source.create("the_name_of_my_source", SourceVisibility.SECURED);
-            System.out.println(String.format("Source creation status: %s", response.statusCode()));
-            System.out.println(String.format("Source creation response: %s", response.body()));
+            HttpResponse<String> response = sourceClient.create("pushy", SourceVisibility.SHARED);
```
Comment (y-lakhdar) [inline samples/CreateSource.java:19]: That file was mainly here to provide examples to users. If you want to perform tests, I recommend doing them in another folder than `samples/`

**Review by y-lakhdar — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (hdhayneCoveo) [reply README.md:23]: you are right i didnt pay attention to that ,  i guess i can't play with any classes public to the client

File: README.md (hunk 5 above)
Comment (hdhayneCoveo) [reply README.md:30]: same reply as above

File: samples/CreateSource.java (hunk 6 above)
Comment (hdhayneCoveo) [reply samples/CreateSource.java:15]: mistake :) i was just playing with it to make sure it is working , good catch , i will put it back

File: samples/CreateSource.java (hunk 7 above)
Comment (hdhayneCoveo) [reply samples/CreateSource.java:19]: totally agree

File: src/main/java/com/coveo/pushclient/ApiStatus.java (hunk 2 above)
Comment (hdhayneCoveo) [reply src/main/java/com/coveo/pushclient/ApiStatus.java:6]: yep i will double check and name it accordingly , thanks

File: src/main/java/com/coveo/source/SourceClient.java (hunk 1 above)
Comment (hdhayneCoveo) [reply src/main/java/com/coveo/source/SourceClient.java:179]: nop , it is IDE  mistake and mine that i missed to double check this one

**Comment by hdhayneCoveo** [comment]
thanks @samisayegh  @y-lakhdar @louis-bompart  for all your comments , i will close this one , since i made a lot of changes in exposed classes making breaking changes for clients already using this sdk 

i have to think how to include catalog source client without touching the existing classes and without duplicating the code  , and will do better with the next PR :)

**Pull request closed without merging**